### DISCUSSION TITLE
running RaftNode example in tcp mode. All nodes become leader

### DISCUSSION THREAD

**Discussion by AntonWerenberg** [dotnet/dotNext#80] [Q&A] [answered]
When attempting to run the unmodified RaftNode example for tcp nodes I encounter problems.
all running are in the leader state and appear not to interact with each other.

Upon closer inspection, I have found that the memberlist of each node only contains itself. I suspect that the problem may be because the node is configured with coldstart = true. Could this be the case?

Upon reading the documentation I am in doubt whether only a single node should be configured with coldstart = true.

When running the example with http configuration everything runs flawlessly.

commands run in seperate teminals when starting example in tcp mode
~/dotNext/src/examples/RaftNode$ dotnet run --framework net6.0 tcp 3262 node1
~/dotNext/src/examples/RaftNode$ dotnet run --framework net6.0 tcp 3263 node2
~/dotNext/src/examples/RaftNode$ dotnet run --framework net6.0 tcp 3264 node3

**Comment by sakno** [comment, marked as answer]
Hi @AntonWerenberg , thanks for bug report. 4.x is beta, so bugs are possible. All further activities will be tracked in #81 .

**Reply by sakno** [reply]
@AntonWerenberg , the issue has been fixed in _RaftNode_ example code. `ColdStart` must be **false** for all nodes. Of course, this is working only for example. In the production, the initial node must be started with `ColdStart=true`.
